inlang/source-code/versioned-interfaces/message-bundle/src/interface.ts

```diff
@@ -0,0 +1,115 @@
+import { LanguageTag as Locale } from "@inlang/language-tag"
+import { Type, type Static } from "@sinclair/typebox"
+
+export type Literal = Static<typeof Literal>
+export const Literal = Type.Object({
+	type: Type.Literal("literal"),
+	value: Type.String(),
+})
+
+/**
+ * A (text) element that is translatable and rendered to the UI.
+ */
+export type Text = Static<typeof Text>
+export const Text = Type.Object({
+	type: Type.Literal("text"),
+	value: Type.String(),
+})
+
+export type VariableReference = Static<typeof VariableReference>
+export const VariableReference = Type.Object({
+	type: Type.Literal("variable"),
+	name: Type.String(),
+})
+
+export type Option = Static<typeof Option>
+export const Option = Type.Object({
+	name: Type.String(),
+	value: Type.Union([Literal, VariableReference]),
+})
+
+export type FunctionAnnotation = Static<typeof FunctionAnnotation>
+export const FunctionAnnotation = Type.Object({
+	type: Type.Literal("function"),
+	name: Type.String(),
+	options: Type.Array(Option),
+})
+
+/**
+ * An expression is a reference to a variable or a function.
+ *
+ * Think of expressions as elements that are rendered to a
+ * text value during runtime.
+ */
+export type Expression = Static<typeof Expression>
+export const Expression = Type.Object({
+	type: Type.Literal("expression"),
+	arg: Type.Union([Literal, VariableReference]),
+	annotation: Type.Optional(FunctionAnnotation),
+})
+
+// export type FunctionReference = {
+// 	type: "function"
+// 	name: string
+// 	operand?: Text | VariableReference
+// 	options?: Option[]
+// }
+
+/**
+ * A pattern is a sequence of elements that comprise
+ * a message that is rendered to the UI.
+ */
+export type Pattern = Static<typeof Pattern>
+export const Pattern = Type.Array(Type.Union([Text, Expression]))
+
+/**
+ * A variant contains a pattern that is rendered to the UI.
+ */
+export type Variant = Static<typeof Variant>
+export const Variant = Type.Object({
+	/**
+	 * The number of keys in each variant match MUST equal the number of expressions in the selectors.
+	 *
+	 * Inspired by: https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#pattern-selection
+	 */
+	// a match can always only be string-based because a string is what is rendered to the UI
+	match: Type.Array(Type.String()),
+	pattern: Pattern,
+})
+
+export type InputDeclaration = Static<typeof InputDeclaration>
+export const InputDeclaration = Type.Object({
+	type: Type.Literal("input"),
+	name: Type.String(),
+
+	//TODO make this generic so that only Variable-Ref Expressions are allowed
+	value: Expression,
+})
+
+export type LocalDeclaration = Static<typeof InputDeclaration>
+export const LocalDeclaration = Type.Object({
+	type: Type.Literal("local"),
+	name: Type.String(),
+	value: Expression,
+})
```

Comment:
Do we need local declarations for v1?

Reply:
Yes, we want to be able to import MF2 using this, right?

Reply:
Why do we want to import MF2 for the MVP? 

- nobody uses MF2
- let's ship only what we need to unblock apps namely (variants) 
- ship in a way to make the data structure future proof

MF2 is still changing. anything we don't need i wouldn't introduce now

Reply:
Here still seems to be some unalignment. Let's achieve consensus @jldec @LorisSigrist @martin-lysk. 

I wouldn't risk adding features from MF2 that apps don't need right now to support variants (pluralization, gendering) as they still change. I don't know why the need to import MF2 is recurringly coming up. We don't need to import MF2 just yet. Nobody uses MF2.

Reply:
Editing nested functions without locals also feels harder from UI/UX perspective, that doing it with locals. Do you agree @NilsJacobsen ?

Reply:
I would go step by step to enable pluralization first. Markup, for example, is unrelated to pluralization. But @jldec this will be your decision now. Being compatible with full MF2 from the get-go is in wrong IMO. The spec will change and we have different requirements because we have apps. 

What is required for pluralization? 

- MessageBundle
- Variants
- Pluralization function 

Seems not to be required: 

- Markup
- Other functions (like date formatting, gendering etc.)
- local declarations

Reply:
Sync notes https://linear.app/opral/issue/MESDK-77/requirements-for-gender-and-plurals#comment-0dd412be. 

It seems like local declarations can be removed if we go the bottoms up approach. cc @martin-lysk

Reply:
👍 Removed for now.